src/main/java/com/atherys/towns/service/PlotService.java

```diff
@@ -14,91 +13,151 @@
+     *          North
+     *
+     * The two points in a PlotSelection are not necessarily the corners we need, we may need to work out opposite corners
+     *
+     * Additionally, as we want the borders around a set of blocks, we need to Floor SW.x, Ceil SW.y, Ceil NE.x, Floor NE.y
+     *
+     * @param plot The plot to populate
+     * @param selection The selection to use
+     */
+    private void populatePlotFromSelection(Plot plot, PlotSelection selection) {
         Vector2d pA = selection.getPointA().getPosition().toVector2(true);
         Vector2d pB = selection.getPointB().getPosition().toVector2(true);
 
         Vector2d pNE;
         Vector2d pSW;
 
-        //              NE
-        //      +---+ pA
-        //      |   |
-        //   pB +---+
-        // SW
+        //             +z
+        //     pB, SW  *
+        //         +---+
+        //         |   |
+        //  -x  <- +---+ -> +X pA
+        //         *    pA NE
+        //         -z
+        //
         if (pA.getX() > pB.getX() && pA.getY() < pB.getY()) {
             pNE = pA;
             pSW = pB;
-
-            //    +---+ pB
-            //    |   |
-            // pA +---+
+        //             +z
+        //     pA, SW  *
+        //         +---+
+        //         |   |
+        //  -x  <- +---+ -> +X pA
+        //         *    pB NE
+        //         -z
+        //
         } else if (pA.getX() < pB.getX() && pA.getY() > pB.getY()) {
             pNE = pB;
             pSW = pA;
-
-            // pB +---+
-            //    |   |
-            //    +---+ pA
+        //         +z
+        //         *     pA
+        //         +---+
+        //         |   |
+        //  -x  <- +---+ -> +X pA
+        //     pB      *
+        //            -z
+        //
         } else if (pA.getX() > pB.getX() && pA.getY() > pB.getY()) {
             pNE = Vector2d.from(pA.getX(), pB.getY());
             pSW = Vector2d.from(pB.getX(), pA.getY());
-
-            // pA +---+
-            //    |   |
-            //    +---+ pB
+        //         +z
+        //         *     pB
+        //         +---+
+        //         |   |
+        //  -x  <- +---+ -> +X pA
+        //     pA      *
+        //            -z
+        //
         } else if (pA.getX() < pB.getX() && pA.getY() < pB.getY()) {
             pNE = Vector2d.from(pB.getX(), pA.getY());
             pSW = Vector2d.from(pA.getX(), pB.getY());
-
         } else {
             throw new IllegalArgumentException("Could not resolve south-west and north-east plot points.");
         }
 
-        plot.setNorthEastCorner(new Vector2i(pNE.getFloorX(), pNE.getFloorY()));
-        plot.setSouthWestCorner(new Vector2i(pSW.getFloorX(), pSW.getFloorY()));
-
-        plot.setName(DEFAULT_PLOT_NAME);
-
-        return plot;
+        plot.setSouthWestCorner(new Vector2i(pSW.getFloorX(), pSW.getFloorY() + 1));
+        plot.setNorthEastCorner(new Vector2i(pNE.getFloorX() + 1, pNE.getFloorY()));
```

Comment:
Is generating a plot or "Rectangle" now something that should be moved to MathUtils?

Reply:
Not sure it fits, unless I make the arguments two points instead of a PlotSelection?

Reply:
I think it might look a bit cleaner that way yeah. Since most of it seems to not really rely on anything but math, but that's just a suggestion. I'm fine either way